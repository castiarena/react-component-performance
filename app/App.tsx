import React from 'react';
import { Switch, Route } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import reset from 'styled-reset';
import customTheme from './theme';
import Test from './pages/test/Test';
import Performance from './pages/performance/Performance';
import Header from './components/Header';
import media from './components/Layout/MediaQueries';
import LazyLoad from './components/LazyLoad';

const MainWrapper = styled.div`
    font-family: 'Manrope', sans-serif;
    max-width: 1280px;
    margin: 0 auto;
    ${media.desktop}{
        padding: ${({ theme }) => theme.spacing.medium};
    }
    ${media.mobile}{
        padding: ${({ theme }) => `${theme.spacing.tiny} ${theme.spacing.tiny} ${theme.spacing.large}`} ;
    }
`;

const GlobalStyles = createGlobalStyle`
    ${reset};
    * {
        box-sizing: border-box;
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
    }
    html, body {
      background-color: #FBFCFF;
    }
`;

const App: React.FC = () => (
    <ThemeProvider theme={customTheme}>
        <MainWrapper>
            <GlobalStyles />
            <Header routes={[{
                to: '/',
                name: 'Home',
            }, {
                to: '/test',
                name: 'Test',
            }, {
                to: '/performance',
                name: 'Performance',
            }]}
            />
            <div>
                <Switch>
                    <Route exact path="/">
                        <LazyLoad importComponent={() => import('./pages/home/Home')} />
                    </Route>
                    <Route path="/test">
                        <LazyLoad importComponent={() => import('./pages/test/Test')} />
                    </Route>
                    <Route path="/performance">
                        <LazyLoad importComponent={() => import('./pages/performance/Performance')} />
                    </Route>
                </Switch>
            </div>
        </MainWrapper>
    </ThemeProvider>
);

export default App;
