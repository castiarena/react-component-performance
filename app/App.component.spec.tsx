import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import 'jest-styled-components';

import App from './App';

describe('App component test suite', () => {
    it('smoke test, snapshot only ', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <App />
            </BrowserRouter>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
