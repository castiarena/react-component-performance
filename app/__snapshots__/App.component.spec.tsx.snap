// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test suite smoke test, snapshot only  1`] = `
<DocumentFragment>
  .c4 {
  display: grid;
  grid-template-columns: repeat(1,1fr);
  grid-column-gap: 40px;
  padding: 20px;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-column-gap: 12px;
  padding: 6px;
}

.c12 {
  border: none;
  padding: 16px 30px;
  border-radius: 10px;
  background-color: #0084ff;
  color: #fff;
  -webkit-transition: background-color .2s ease-out;
  transition: background-color .2s ease-out;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12:hover {
  cursor: pointer;
  background-color: #a7c3dc;
}

.c12:active,
.c12:focus {
  outline: none;
}

.c5 {
  font-size: 2em;
  margin: 26px 0;
  text-align: center;
  color: #0A0A33;
  font-weight: bold;
}

.c8 {
  font-size: 1.3em;
  margin: 26px 0;
  text-align: left;
  color: #0A0A33;
  font-weight: bold;
}

.c9 {
  font-size: 1em;
  margin: 12px 0;
  line-height: 1.3em;
  text-align: left;
  color: #0A0A33;
  font-weight: normal;
}

.c10 {
  color: #0084ff;
}

.c10:hover {
  color: #a7c3dc;
}

.c7 {
  padding: 26px;
  border-radius: none;
  box-shadow: none;
  background-color: #FBFCFF;
}

.c11 {
  padding: 26px;
  border-radius: 20px;
  box-shadow: 0 4px 12px -1px rgba(0,0,0,.1);
  background-color: #FBFCFF;
}

.c2 {
  color: #0084ff;
}

.c2:hover {
  color: #a7c3dc;
}

.c1 {
  padding: 30px;
  display: grid;
  grid-template-columns: repeat(4,1fr);
  grid-column-gap: 16px;
  justify-items: center;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 16px;
}

.c0 {
  font-family: 'Manrope',sans-serif;
}

@media (max-width:576px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:576px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div
    class="c0"
  >
    <header
      class="c1"
    >
      <img
        alt="Example"
        src="./assets/logo-realworldone.svg"
      />
      <a
        class="c2 c3"
        href="/"
      >
        Home
      </a>
      <a
        class="c2 c3"
        href="/test"
      >
        Test
      </a>
      <a
        class="c2 c3"
        href="/performance"
      >
        Performance
      </a>
    </header>
    <div>
      <div
        class="c4"
      >
        <h1
          class="c5"
        >
          Real world one
          <strong>
            Frontend Developer Test
          </strong>
        </h1>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <h2
              class="c8"
            >
              Prerequisites
            </h2>
            <p
              class="c9"
            >
              Choose any JSON dummy data online generator. This will be your data source that will be used within your application (so called Test Data).
            </p>
            <p
              class="c9"
            >
              The Test Data size should be at least 500 records. e.g.
              <a
                class="c10"
                href="https://jsonplaceholder.typicode.com/comments"
                target="_blank"
              >
                https://jsonplaceholder.typicode.com/comments
              </a>
            </p>
          </div>
          <div
            class="c11"
          >
            <h2
              class="c8"
            >
              Tasks
            </h2>
            Can see the full tasks on the PDF file:
            <p
              class="c9"
            >
              <a
                class="c12"
                download=""
                href="./assets/instructions.pdf"
              >
                Download
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
